source/common/network/cidr_range.h

```diff
@@ -117,6 +120,23 @@ class CidrRange {
   int length_;
 };
 
+/**
+ * Class for keeping a list of CidrRanges, and then determining whether an
+ * IP address is in the CidrRange list.
+ */
+class IpList {
+public:
+  IpList(const std::vector<std::string>& subnets);
+  IpList(const Json::Object& config, const std::string& member_name);
```

Comment:
Constructor is used here: https://github.com/lyft/envoy/blob/master/source/common/filter/auth/client_ssl.cc#L29. I think constructing from the json object does make sense. But changing to a static factory isn't much work.